pages/admin/finance/index.tsx

```diff
@@ -2,11 +2,40 @@ import Chart from 'chart.js/auto';
 import { useRef, useEffect } from 'react';
 import { useSession } from 'next-auth/react';
 import Loading from 'components/Loading';
+import { Query, dehydrate, useQuery } from 'react-query';
+import { GetServerSideProps } from 'next';
+import { gqlQueries, queryClient } from 'src/api';
+
+export const getServerSideProps: GetServerSideProps = async (ctx) => {
+  await queryClient.prefetchQuery(['getFinanceData'], () => gqlQueries.getFinanceData());
+  return {
+    props: {
+      // dehydratedState: dehydrate(queryClient),
+    },
+  };
+};
 
 export default function financePage() {
```

Comment:
Can we have restriction to make sure that only officers have access using [this context](https://github.com/acmutd/portal-next/blob/next/pages/admin/officer/add.tsx#L16) and the following [check](https://github.com/acmutd/portal-next/blob/next/pages/admin/officer/add.tsx#L54-L56)?